### PR TITLE
Popup window

### PR DESCRIPTION
**Pull request by Maxwell011**
Implemented the following interactions:

When the user clicks (or taps) the button to check project details, the popup with details about the project appears.
When the user clicks (or taps) the close (X) button, the popup disappears.
In order to associate each project with the popup window details, I refactor the project section:

- You need to use a JavaScript array to store all of the information for all projects.
- For each project you need to store the following pieces of data in a JavaScript object, at a minimum: name, description,  featured image, technologies, link to live version, and link to the source.
I updated the main page so the projects section is created dynamically using the information stored in that JavaScript object. 
After that, I implement the popup window.
I implement popups for both mobile and desktop screen sizes:
- Mobile popup
- Desktop popup

Kindly Review 🙏

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -104,6 +104,12 @@ <h1 class="heading">
...
 
+      <!-- The Modal -->
+      <section
+        id="myModal"
+        class="modal modal-dialog modal-dialog-centered modal-dialog-scrollable"
+      ></section>
+
       <!-- Works/projects section-->
       <section class="grid-container">
         <h2 class="heading-container heading-one" id="portfolio">
```
Comment (thecodechaser) [inline index.html:115]: - [ ] The works section should be created dynamically using the information stored in that JavaScript object as requested in the project requirements. Please update the works section and make sure that all of the HTML in that section is created when the page loads. `(You should delete the works section from the main page and create it in the javascript file dynamically).`

image

File: popsup.js [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+    <h3 id="title">${project.name}</h3>
+  </div>
+  <ul class="tags" id="tech">
+  ${project.technologiesDesktop.map((ele) => `<li>${ele}</>`)}
+  </ul>
+  <ul class="tags" id="mobile-ul">
+  ${project.technologiesMobile.map((ele) => `<li>${ele}</>`)}
+  </ul>
+  <p id="description">${project.description}</p>
+  <div class="modal-footer">
```
Comment (thecodechaser) [inline popsup.js:59]: - [ ] On the desktop screen the project description is overflowing from the model, Please fix this issue and make sure that the description is displayed inside the model and your design matches with the [Figma](https://www.figma.com/file/l7SqJ3ZfkAKih9sFxvWSR4/Microverse-Student-Project-1?node-id=1%3A468) design.

### Your design
image

### Figma design
image

File: popsup.js [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+      /></button
+  ></a>
+  <a href="${project.sourceLink}">
+    <button type="submit" class="project-view-btn">
+      <span>See Source</span>
+      <img
+        src="${project.githubIcon}"
+        alt="See Source Code"
+      /></button
+  ></a>
```
Comment (thecodechaser) [inline popsup.js:75]: - [ ] Please remove the black border from the buttons and make sure the button's texts have equal padding from all sides.

### Your design
image

### Figma design
image

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @Maxwell011👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- GitHub flow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Maxwell011** [comment]
I have added the needed changes asked by the reviewer 👍 

- The Work section has been created Dynamically 
- I have increased the height.
- I have removed the border.

Kindly Review 🙏

**Review by victorokoroji — APPROVED** [review]
Hi @Maxwell011, 

# ✔️   Status: APPROVED 🏆 🎉 🎊 

You have done a great job on this project, keep it up 👍 🥇. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
congratulations!  🎉💯🌟 

## Top highlight 🎯

- The Pull Request is following the GitHub Flow ✅ 
- The PR title and description are professional and descriptive ✅ 
- The commit messages are meaningful ✅  
- Linters are working properly ✅ 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**